core/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -409,9 +399,9 @@ public void readFrom(StreamInput in) throws IOException {
         maxRetries = in.readVInt();
         requestsPerSecond = in.readFloat();
         if (in.getVersion().onOrAfter(Version.V_5_1_1)) {
-            slices = in.readVInt();
+            slices = new Slices(in);
         } else {
-            slices = 1;
+            slices = Slices.DEFAULT;
         }
```

Comment:
I'm not sure how backwards compatibility should work here. This change uses `-1` as the value here for serializing the auto setting, which won't be a valid slices int value in earlier versions. Something like

```
readFrom {
  if version on or after 5.1.1 {
    slices = new Slices(in)
  } else {
    slices = default
  }
}

writeTo {
  if version on or after 5.1.1 and before 6.1.0 {
    if slices is auto {
      throw exception
    } else {
      slices.writeTo(out)
    }
  } else if version on or after 6.1.0
    slices.writeTo(out)
  } else {
    if slices > 1 or slices is auto, throw exception
  }
}